test/controllers/responses_controller_test.rb

```diff
@@ -1,7 +1,62 @@
 require 'test_helper'
 
 class ResponsesControllerTest < ActionDispatch::IntegrationTest
-  # test "the truth" do
-  #   assert true
-  # end
+  include Devise::Test::IntegrationHelpers
+
+  setup do
+    @user     = create(:user)
+    @admin    = create(:admin)
+    @survey   = create(:survey, user: @user)
+    @response = create(:response, survey: @survey)
+  end
+
+  test "response can only be destroyed by a logged in user" do
+    assert_difference('Response.count', 0) do
+      delete survey_response_url(@survey, @response)
+    end
+    assert_redirected_to new_user_session_path
+  end
+
+  test "response can only be deleted by owner of the survey" do
+    @survey2   = create(:survey, user: @admin)
```

Comment:
No need to make these instance variables if they're not being accessed anywhere else, we can just call them `survey` and they will not be available outside the scope of this test

Reply:
Many thanks @spencerldixon